src/main/java/book/store/onlinebookstore/controller/BookController.java

```diff
@@ -21,37 +24,48 @@
 @RestController
 @RequiredArgsConstructor
 @RequestMapping("/api/books")
+@Tag(name = "Book management", description = "Endpoints for managing books")
 public class BookController {
     private final BookService bookService;
 
     @GetMapping("/{id}")
+    @Operation(summary = "Get book by id", description = "Get the book by its id number")
     public BookDto getBookById(@PathVariable Long id) {
         return bookService.findById(id);
     }
 
     @GetMapping
-    public List<BookDto> getAll() {
-        return bookService.findAll();
+    @Operation(summary = "Get all books",
+            description = "Get all books per website pages, default: page = 0, size = 20")
+    public List<BookDto> getAll(Pageable pageable) {

```

Comment:
shouldn't you add `@PageableDefault(page = 0, size  = 20)` to your request parameter?